### ISSUE TITLE
___HyperlinkRegistry.hyperlinks problem

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dbarra/xdocreport#79]
```
Hi:

I use @before-row[#list vos as vo]${vo.href}@after-row[/#list] in my template 
file.
when I put an empty vos in IConext ,there're some exception will be throw like :

Expression ___HyperlinkRegistry.hyperlinks is undefined on line 1, column 1482 
in 
fr.opensagres.xdocreport.document.docx.DocXReport@427b7b5d!word/_rels/document.x
ml.rels.
The problematic instruction:
----------
==> list ___HyperlinkRegistry.hyperlinks as ___info [on line 1, column 1475 in 
fr.opensagres.xdocreport.document.docx.DocXReport@427b7b5d!word/_rels/document.x
ml.rels]
----------

Java backtrace for programmers:
----------
freemarker.core.InvalidReferenceException: Expression 
___HyperlinkRegistry.hyperlinks is undefined on line 1, column 1482 in 
fr.opensagres.xdocreport.document.docx.DocXReport@427b7b5d!word/_rels/document.x
ml.rels.
    at freemarker.core.TemplateObject.assertNonNull(TemplateObject.java:125)
    at freemarker.core.IteratorBlock.accept(IteratorBlock.java:100)
    at freemarker.core.Environment.visit(Environment.java:221)
    at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:79)
    at freemarker.core.Environment.visit(Environment.java:221)
    at freemarker.core.MixedContent.accept(MixedContent.java:92)

Is the vos can't be empyt? 
```

Original issue reported on code.google.com by `x.resolute@gmail.com` on 20 Feb 2012 at 4:59

Attachments:
- [test.docx](https://storage.googleapis.com/google-code-attachments/xdocreport/issue-79/comment-0/test.docx)

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Which version do you use of XDocReport? 0.9.5? If no, could you test with this 
version? Since 0.9.5, the key ___HyperlinkRegistry doesn't exist. If you are 
using 0.9.5, could you attache too your Java code which create IContext.

Thanks.

Regards Angelo
```

Original comment by `angelo.z...@gmail.com` on 20 Feb 2012 at 10:06
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I use 0.9.4.Is it in 0.95 is ok for this case?
sometimes I define some mergeFields in the template file but in the out file 
these fields are still mergeField 
Are there something wrong?
--------------------------------------------------------------------------------
------------
Here an other error log when I run junit in a web appliaction ,The error like :
[Fatal Error] :1:1: Content is not allowed in prolog.
错误：“Content is not allowed in prolog.”
错误：“Content is not allowed in prolog.”
错误：“com.sun.org.apache.xml.internal.utils.WrappedRuntimeException: 
Content is not allowed in prolog.”
错误：“Content is not allowed in prolog.”
But the same template file when I run it in a j2SE application ,it is ok.



--------------------------------------
sorry fro my english

```

Original comment by `x.resolute@gmail.com` on 20 Feb 2012 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
>I use 0.9.4.Is it in 0.95 is ok for this case?
Please try O.9.5 because I have refactored a lot HyperlinkRegistry. 

>sometimes I define some mergeFields in the template file but in the out file 
these >fields are still mergeField 
>Are there something wrong?
Please provide your docx.

> Here an other error log when I run junit in a web appliaction ,The error like 
:
> [Fatal Error] :1:1: Content is not allowed in prolog.
It seems that you have activated log. When it is activated, XML entry are 
displayed in the console and we do pretty print xml. In O.9.4, there is a bug 
with pretty print with Freemarker. So please use 0.9.5.

Regards Angelo


```

Original comment by `angelo.z...@gmail.com` on 20 Feb 2012 at 10:59

**Comment by GoogleCodeExporter** [comment]
```
HI:
when I use  0.95 the log still out in console .
other propleam is fiexd;
Thanks.
```

Original comment by `x.resolute@gmail.com` on 20 Feb 2012 at 11:36

Attachments:
- [log4j.xml](https://storage.googleapis.com/google-code-attachments/xdocreport/issue-79/comment-4/log4j.xml)

**Comment by GoogleCodeExporter** [comment]
```
For the log, excuse me it will be fixed in the next release 0.9.6.
Regards Angelo
```

Original comment by `angelo.z...@gmail.com` on 20 Feb 2012 at 12:45

**Comment by GoogleCodeExporter** [comment]
Original comment by `angelo.z...@gmail.com` on 13 Mar 2012 at 2:43
- Changed state: **Fixed**
